src/pysweepme/DeviceManager.py

```diff
@@ -58,43 +86,28 @@ def get_device(name, folder=".", port_string=""):
             It is required if the driver connects to an instrument and needs to open a specific port.
 
     Returns:
-        Device object of the driver
+        Initialized Device object of the driver with port and default parameters set.
     """
-
     if folder == "":
         folder = "."
     if name.startswith(os.sep):
         name = name[1:]
     if name.endswith(os.sep):
         name = name[:-1]
 
-    try:
-        # Loads .py file as module
-        module = imp.load_source(name, get_main_py_path(folder + os.sep + name))
-    except:
-        error()
-        raise Exception(f"Cannot load Driver '{name}' from folder {folder}. Please change folder or copy Driver "
-                        "to your project.")
- 
-    device = module.Device()
-    device._import_path = module.__file__  # needs a comment
-    
+    device = instantiate_device(folder, name)
+
     if port_string != "":
         if device.port_manager:
             port = get_port(port_string, device.port_properties)
             device.port = port
```

Comment:
One could use the set_port method of EmptyDevice here, but can be also changed later

Reply:
done